avm/src/daemon/ctrl.rs

```diff
@@ -124,6 +140,66 @@ async fn control_port(req: Request<Body>) -> Result<Response<Body>, Infallible>
   }
 }
 
+async fn handle_start(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+  // Receive POST and save daemon properties
+  match get_daemon_props(req).await {
+    Ok(_) => Ok(Response::builder().status(200).body("ok".into()).unwrap()),
+    Err(err) => {
+      error!(ErrorType::DaemonStartFailed, "{:?}", err).await;
+      Ok(Response::builder().status(500).body("fail".into()).unwrap())
+    }
+  }
+}
+
+async fn get_daemon_props(req: Request<Body>) -> DaemonResult<()> {
+  let bytes = body::to_bytes(req.into_body()).await?;
+  let body: DaemonProperties = serde_json::from_slice(&bytes).unwrap();
+  maybe_dump_files(&body)?;
+  DAEMON_PROPS
+    .set(DaemonProperties {
+      clusterId: body.clusterId,
+      agzB64: body.agzB64,
+      deployToken: body.deployToken,
+      domain: body.domain,
+      DockerfileB64: body.DockerfileB64,
+      appTarGzB64: body.appTarGzB64,
+      envB64: body.envB64,
```

Comment:
I would like to make a suggestion on these three properties.

Can we make them sub-properties on a `filesB64` object where the key is literally just the name of the file and the value is the base64-encoded data? Then we can generalize all of this to just: "here are the files you need to put on disk during startup for whatever reason needed by the `agz` payload".

This will let us eliminate all of the special hackery within the AVM for Anycloud and eventually give the power to include static files to anyone running `alan deploy new` / `anycloud new`.

Reply:
Good idea, makes sense to me